test/Microsoft.AspNet.Mvc.IntegrationTests/BodyValidationIntegrationTests.cs

```diff
@@ -42,11 +42,18 @@ private class Address
                 ParameterType = typeof(Person)
             };
 
-            var operationContext = ModelBindingTestHelper.GetOperationBindingContext();
-            var httpContext = operationContext.HttpContext;
-
-            ConfigureHttpRequest(httpContext.Request, string.Empty);
-            var modelState = new ModelStateDictionary();
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext(
+              request =>
+              {
+                  request.Body = new MemoryStream(Encoding.UTF8.GetBytes("{ \"Id\":1234 }"));
+                  request.ContentType = "application/json";
+              });
+
+            var actionContext = operationContext
+                .HttpContext
+                .RequestServices
+                .GetRequiredService<IScopedInstance<ActionContext>>().Value;
+            var modelState = actionContext.ModelState;
```

Comment:
As an alternative can we just have the tests initialize things in the order they need to go?

1). Create HttpContext
2). Create ActionContext
3). Create OperationBindingContext (forces service initialization)